action/receipt.go

```diff
@@ -145,6 +149,17 @@ func (receipt *Receipt) AddTransactionLogs(logs ...*TransactionLog) *Receipt {
 	return receipt
 }
 
+// ExecutionRevertMsg returns the list of execution revert error logs stored in receipt.
+func (receipt *Receipt) ExecutionRevertMsg() string {
+	return receipt.executionRevertMsg
+}
+
+// AddExecutionRevertMsg add executionerrorlogs to receipt and filter out "" log.
+func (receipt *Receipt) AddExecutionRevertMsg(revertReason string) *Receipt {
+	receipt.executionRevertMsg = revertReason
```

Comment:
add condition  if receipt.executionRevertMsg == ""?

Reply:
what do you want to handle in this condition?

Reply:
i mean
```go
if receipt.executionRevertMsg == "" {
   receipt.executionRevertMsg = revertReason
}
```
so it captures the first error string, and won't be overwritten on later calls, thought that's not the case (AddExecutionRevertMsg is only called once in your code change)

Reply:
and better 
```go
if receipt.executionRevertMsg == "" &&  revertReason != "" {
   receipt.executionRevertMsg = revertReason
}
```

Reply:
thanks for the clarification!